src/payment/payment-strategy-registry.ts

```diff
@@ -46,12 +51,9 @@ export default class PaymentStrategyRegistry extends Registry<PaymentStrategy, P
         }
 
         if (paymentMethod.gateway === PaymentStrategyType.CHECKOUTCOM) {
-            switch (paymentMethod.id) {
-                case 'credit_card':
-                    return PaymentStrategyType.CHECKOUTCOM;
-                default:
-                    return PaymentStrategyType.CHECKOUTCOM_APM;
-            }
+            return paymentMethod.id in checkoutcomStrategies
+                ? checkoutcomStrategies[paymentMethod.id as checkoutcomMethods]
```

Comment:
to avoid the casting, maybe set the proper type for `checkoutcomStrategies`?
```js
const checkoutcomStrategies: {[key in keyof typeof checkoutcomMethods]: PaymentStrategyType}